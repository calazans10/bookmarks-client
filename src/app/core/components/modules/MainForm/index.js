import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const MainForm = ({ legend, children, onSubmit }) => (
  <form autoComplete="off" noValidate className="main-form" onSubmit={onSubmit}>
    <fieldset>
      <legend className="main-form__legend">{legend}</legend>
      {children}
    </fieldset>
  </form>
);

MainForm.propTypes = {
  legend: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default MainForm;
