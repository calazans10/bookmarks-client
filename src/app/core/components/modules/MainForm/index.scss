@import 'styles/utilities';

.main-form {
  @media(min-width: 600px) {
    width: rem(522);
    margin: rem(60) auto;
  }

  @media(min-width: 800px) {
    margin-top: rem(100);
    margin-bottom: rem(100);
  }

  @media(min-width: 1044px) {
    width: rem(600);
  }

  &__legend {
    margin-bottom: rem(25);
    font-size: em(24);
    font-weight: 500;
  }

  a {
    margin-left: rem(10);
  }
}
