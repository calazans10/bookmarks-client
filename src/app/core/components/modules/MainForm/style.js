import styled from 'styled-components';
import { em, rem } from 'polished';

export const Container = styled.form`
  @media (min-width: 600px) {
    width: ${rem('522px')};
    margin: ${rem('60px')} auto;
  }

  @media (min-width: 800px) {
    margin-top: ${rem('100px')};
    margin-bottom: ${rem('100px')};
  }

  @media (min-width: 1044px) {
    width: ${rem('600px')};
  }

  a {
    margin-left: ${rem('10px')};
  }
`;

export const Legend = styled.legend`
  margin-bottom: ${rem('25px')};
  font-size: ${em('24px')};
  font-weight: 500;
`;
