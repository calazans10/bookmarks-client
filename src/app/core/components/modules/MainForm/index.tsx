import React from 'react';
import { Container, Legend } from './style';

type MainFormProps = {
  legend: string;
  children: React.ReactNode;
  onSubmit: () => void;
};

const MainForm = ({ legend, children, onSubmit }: MainFormProps) => (
  <Container autoComplete="off" noValidate onSubmit={onSubmit}>
    <fieldset>
      <Legend>{legend}</Legend>
      {children}
    </fieldset>
  </Container>
);

export default MainForm;
