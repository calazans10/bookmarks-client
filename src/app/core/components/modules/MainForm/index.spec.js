import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainForm from './index';

describe('MainForm', () => {
  it('renders without crashing', () => {
    const props = {
      legend: 'A legend',
      onSubmit: jest.fn(),
      children: (
        <div>
          <label htmlFor="test">A label</label>
          <input id="test" name="test" type="text" />
        </div>
      ),
    };
    const wrapper = shallow(<MainForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
