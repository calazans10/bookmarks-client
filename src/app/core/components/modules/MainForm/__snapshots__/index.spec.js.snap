// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainForm renders without crashing 1`] = `
<form
  autoComplete="off"
  className="main-form"
  noValidate={true}
  onSubmit={[MockFunction]}
>
  <fieldset>
    <legend
      className="main-form__legend"
    >
      A legend
    </legend>
    <div>
      <label
        htmlFor="test"
      >
        A label
      </label>
      <input
        id="test"
        name="test"
        type="text"
      />
    </div>
  </fieldset>
</form>
`;
